### ISSUE TITLE
[1.12.2] Fill doesn't do anything even though it says so

### ISSUE THREAD

**Issue by LiskoSlayer63** [abused/World-Border#2]
I tried to pre-generate my world with the `wb fill` command, but it doesn't do anything. Console says that it fills, but if I try to render the world with dynmap it shows nothing except those chunks which were already generated by exploring. Save folder size doesn't grow either, I'm using a round border with 5000 block radius and after fill it's size is only 20MB.

**WorldBorder version:** 1.0.6
**Forge version:** 14.23.1.2561
**SpongeForge version:** 2555-7.0.0-BETA-2764
**Minecraft version:** 1.12.2

**Comment by abused** [comment]
yes i'm aware of this, was the reason i have not released to curse yet, i hot fixed issue #1 and am working on this one as well

**Comment by smmmadden** [comment]
I too am running into the same issue where I run the wb world fill command and it runs and runs, but the map never updates.  Here is my setup:
Minecraft: 1.12.2
SpongeAPI: 7.0.0-SNAPSHOT-0c11218
SpongeForge: 1.12.2-2555-.0.0-BETA-2758
Minecraft Forge: 14.23.1.2555
World-Border: 1.12.2-1.0.6

**Comment by smmmadden** [comment]
I just tried again with the latest builds: forge-1.12.2-14.23.1.2587 and spongeforge-1.12.2-2555-7.1.0-BETA-2831 and it is reproducible there as well.  It fixed another issue I had with worldedit, so was hoping it would do both. :-)

**Comment by abused** [comment]
I apologize for not being able to push out an update, I've been very busy irl though I should have time this weekend to fix it, I have detected the issue so hopefully it'll be an easy fix!

**Comment by smmmadden** [comment]
no worries, I appreciate the effort it takes to fix it. :-)

**Comment by smmmadden** [comment]
just curious @abused is this mod using the standard worldborder.* permission nodes?  I'm not seeing them registered in LuckPerms Sponge.  Perhaps this has something to do with the above?
If I try using /wb I get a message that it Opens a crafting table window so users can craft (a minecraft command).  I read also that /worldborder is also a minecraft command so is it possible that the permissions are not being registered for world-border 1.0.6 and that all we are doing is calling the vanilla world border?  It's a stretch, but thought worth asking anyway. :-)

**Comment by Spagz** [comment]
Any update on the fix for the /fill not working?

**Comment by smmmadden** [comment]
Seeting the wb alias to false in the config/nucleus/commands.conf will allow the /wb command to use the world border mod's command.
![image](https://user-images.githubusercontent.com/10309346/35180240-6600e552-fd7a-11e7-8ea3-9429a66116fe.png)
and it will still run the job to update the map, but it doesn't change the underlying issue with the writing of the map updates at least for 1.12.2.

**Comment by smmmadden** [comment]
Hi @abused just checking in to get an update.  Do you have an ETA on when the fix will be ready?
Thanks -Steve

**Comment by smmmadden** [comment]
sorry for the delay, been a busy week at work.  I finally had some time to download onto my test server and swap out 1.0.6 for the 1.0.7.  I re-ran the /wb world fill and I'm not seeing any changes to the map at all.  It's 49% done currently and nothing has changed yet, so I'll wait until it finishes.
I do notice a warning for "Can't keep up! Did the system time change, or is the server overloaded? Running 25938ms behind, skipping 518 tick(s).  I've got more than 2gb free and nothing else running so it seems like a false positive warning.  Should I have removed it and recreated the border or done something else?

**Comment by LiskoSlayer63** [comment]
@summmadden It lags when it generates the world, that's normal I think. I had 8gb allocated RAM and it spammed that warning all the time but it does it only when it's generating the world. The map doesn't show anything before you do a fullrender for dynmap, but don't do this until the fill is complete. 

Personally I first do the fill, then reboot after it finishes to be sure that it saves the world and frees the memory. Then I do the fullrender and after that I reboot again just to be sure. My map have a 5000 block radius and it took about one day to fully complete all this.

**Comment by smmmadden** [comment]
thanks - sounds like a pretty sound approach to me.  I'll repeat the same and let you know.  :-)

**Comment by smmmadden** [comment]
did a restart after the wb fill and then a fullrender.  I'm now seeing the rest of the world finally getting filled in. :-)   SUCCESS!  Thank you very much.  -Steve

**Comment by LiskoSlayer63** [comment]
Glad if I helped!

**Comment by smmmadden** [comment]
while my test server appears to have worked, the prod server which is larger has crashed 3 times after starting the wb fill process.  I've bumped the memory up to 6GB but still not sure why it keeps crashing.

This is the latest.  Each time it shuts the server down and have to type in the wb fill again.
https://pastebin.com/FFudNXiG  Any ideas?

**Comment by LiskoSlayer63** [comment]
@smmmadden It doesn't like SpongeForge, it doesn't even fill it fully when they're enabled on the same time. This was mentioned on the download page. I use SpongeForge too so I just disabled it until the fill was complete, after that it can be enabled again.

**Comment by smmmadden** [comment]
I've removed the sponge forge mod and running the fill command now.  After it writes out a few lines I get an [worldborder]: Error while populating chunk!  and either it follows with a java.lang.NullPointerException or a java.util.ConcurrentModificationException.  It's still holding on and running, just not liking the errors coming out of com.wimbli.WorldBorder.task.WorldFillTask:onServerTick:367

Is this something you know about also?

**Comment by smmmadden** [comment]
All Set - This bug can be closed.
After setting the shape and world's border radius:
1. Shut down the server
2. Remove any sponge mod from the mods folder of the server
3. Remove the 1.0.6 version of World-Border jar
4. Save the 1.0.7 version of the World-Border.jar in the server mods folder
5. Start the server
6. Run the /wb world fill & /wb fill confirm commands
7. Once done, shut down the server
8. Start the server
9. Run the /dynmap fullrender world command
10. Once done, shut down the server
11. Put the sponge mod back into the server mods folder
12. Start the server
13. World should be fully generated to the world border

Thank you very much for the great support and how to get the world regenerated.  I hope these steps help others to do the same. :-)  -Steve

**Comment by LiskoSlayer63** [comment]
@smmmadden I might know what caused the chunk populating error, I'll look it closer when I have time. I'll also add your tutorial steps to the download page!

**Comment by Misteralex007** [comment]
@smmmadden I'm playing with a modpack (FTB Revelation), so what should I do? I just add Dynmap-Forge & World-Border (the beta version with the fix for the illegal generation crash). There are no sponge mod, so I didn't need to remove any mod in theoretical, right?

I don't know if the world is generating actually. I'm waiting for the 100% then I will restart the server to see, but I think it won(t work.

**Comment by smmmadden** [comment]
I've been trying to find where the 1.0.7 download is located and everywhere I checked it shows 1.0.5 or 1.0.6 as the latest version.  Can someone provide a link to where it is and possibly why it isn't available on CurseForge?  Thanks!

**Comment by LiskoSlayer63** [comment]
@smmmadden It's located here:

## **[Unofficial 1.0.7 BETA](https://github.com/LiskoSlayer63/World-Border/releases/tag/1.0.7-beta)**

**Comment by smmmadden** [comment]
awesome!  Thank you. :-)
